src/arch/riscv/pagetable_walker.cc

```diff
@@ -490,8 +490,29 @@ Walker::WalkerState::recvPacket(PacketPtr pkt)
             Addr paddr = walker->tlb->translateWithTLB(vaddr, satp.asid, mode);
             req->setPaddr(paddr);
             walker->pma->check(req);
+
+            if (pmode != PrivilegeMode::PRV_M){
+                bool pass = walker->pmp->pmp_check(req, mode);
+                // what else to do here?
+                // raise some kind of an exception
+
+                if (!pass)
+                {
+                    ExceptionCode code;
+                    if (mode == TLB::Read)
```

Comment:
Actually, I wonder if this code should be factored out into a function (e.g., `createMemException(TLB::Mode)`)

Reply:
created a function